### DISCUSSION TITLE
How to implement user upload stl file to server？

### DISCUSSION THREAD

**Discussion by chaoyangxiao** [Kitware/trame#377] [General] [answered]
Hi，bro.
I want to implement the ability for a user to upload a file to the server and read and render it with vtk.
I've learned the example file_upload so far, but this doesn't meet my needs, it seems to just read the stl information, the stl file is not uploaded to the server side. I hope you can help me, thank you very much!

server = get_server()
with SinglePageWithDrawerLayout(server) as layout:
    with layout.toolbar:
        vuetify.VFileInput(
            v_model=("file_exchange", None),
        )

@server.state.change("file_exchange")
def handle(file_exchange, **kwargs):
    file = ClientFile(file_exchange)

    print(file.info)

if __name__ == "__main__":
    server.start()

**Comment by jourdain** [comment, marked as answer]
Replace `print(file.info)`

by 

```python
from pathlib inport Path

my_file = Path(__file__).with_name(file.name)
my_file.write_binary(file.content)
# now you have the file on the server
```

**Reply by chaoyangxiao** [reply]
Thank you for your reply. I implemented the upload function. Thanks a lot.
By the way, if the server is windows, please use the reference code below for others:

```
@server.state.change("file_exchange")
def handle(file_exchange, **kwargs):
    global file
    file = ClientFile(file_exchange)

def save():
    save_path = Path(r'D:/**/**')  # path
    my_file = save_path / file.name  
    my_file.write_bytes(file.content)  # use write_bytes
```
